File: frontend/src/model/product-category.ts
import {Product} from './product';

export interface ProductCategory {
  id: number,
  title: string,
  description: string,
  coverImage: string,
  coverVideo: string,
  products: Product[]
}

export interface ProductCategoryMap {
  id: number,
  title: string,
  description: string,
  coverImage: string,
  coverVideo: string,
  products: {[key: string]: Product}
}

File: src/main/java/am/techmock/shop/model/ProductCategory.java
package am.techmock.shop.model;

import java.util.Map;

public record ProductCategory(
		int id,
		String title,
		String description,
		String coverImage,
		String coverVideo,
		Map<Integer, Product> products
) {
	static public ProductCategory of(
			int id, String title, String description,
			String coverImage, String coverVideo, Map<Integer, Product> products
	) {
		return new ProductCategory(id, title, description, coverImage, coverVideo, products);
	}
}
